source/sn_coap_protocol.c

```diff
@@ -1207,67 +1212,91 @@ static void sn_coap_protocol_linked_list_blockwise_payload_store(struct coap_s *
         return;
     }
 
-    coap_blockwise_payload_s *stored_blockwise_payload_ptr = NULL;
-
-    /* * * * Allocating memory for stored Payload  * * * */
-
-    /* Allocate memory for stored Payload's structure */
-    stored_blockwise_payload_ptr = handle->sn_coap_protocol_malloc(sizeof(coap_blockwise_payload_s));
+    coap_blockwise_payload_s *stored_blockwise_payload_ptr = sn_coap_protocol_linked_list_blockwise_search(handle, addr_ptr, token_ptr, token_len);
 
-    if (stored_blockwise_payload_ptr == NULL) {
-        tr_error("sn_coap_protocol_linked_list_blockwise_payload_store - failed to allocate blockwise!");
-        return;
-    }
+    if (stored_blockwise_payload_ptr && stored_blockwise_payload_ptr->use_size1) {
+        memcpy(stored_blockwise_payload_ptr->payload_ptr + (block_number * handle->sn_coap_block_data_size), payload_ptr, payload_len);
+    } else if (stored_blockwise_payload_ptr) {
+        uint16_t new_len = stored_blockwise_payload_ptr->payload_len + payload_len;
+        tr_debug("sn_coap_protocol_linked_list_blockwise_payload_store - reallocate from %d to %d", stored_blockwise_payload_ptr->payload_len, new_len);
 
+        stored_blockwise_payload_ptr->payload_ptr = realloc(stored_blockwise_payload_ptr->payload_ptr, new_len);
```

Comment:
realloc() is now bypassing the sn_coap_protocol_alloc and sn_coap_protocol_free -pair.

Reply:
True, that needs to be fixed.

Reply:
Fixed.